nodeup/pkg/model/context.go

```diff
@@ -554,7 +544,8 @@ func EvaluateHostnameOverride(hostnameOverride string) (string, error) {
 	if len(result.Reservations[0].Instances) != 1 {
 		return "", fmt.Errorf("too many instances returned for the single instance-id")
 	}
-	return *(result.Reservations[0].Instances[0].PrivateDnsName), nil
+	hostname := strings.Split(*(result.Reservations[0].Instances[0].PrivateDnsName), ".")[0]
```

Comment:
Have you verified that this function returns the internal FQDN and not just the hostname?

Reply:
yes, i rolled out it locally. With completely removed hostnameOverride in kubelet `kubectl describe node` return node with label `kubernetes.io/hostname=ip-<xxx>-<xxx>-<xxx>-<xxx>`. So protokube should query node by `ip-<xxx>-<xxx>-<xxx>-<xxx>` which is element 0 if we split FQDN by `'.'`.

Reply:
protokube now uses node name instead of `kubernetes.io/hostname` annotation. So this can be removed.